include/BSMPT/gravitational_waves/gw.h

```diff
@@ -0,0 +1,196 @@
+  double fPeakTurbulence       = false; // peak frequency for turbulence
+  double h2OmegaPeakTurbulence = false; // peak amplitude for turbulence
+
+  std::string status = "not_set"; // gw calculation status
+};
+
+class GravitationalWave
+{
+private:
+public:
+  GravitationalWaveData data;
+  GravitationalWave(BounceSolution &BACalc,
+                    const int &which_transition_temp = 3);
+  ~GravitationalWave();
+
+  /**
+   * @brief CalcEpsTurb calculate epsilon for turbulence contribution
+   * @param epsturb_in is the input value for epsturb. If [0..1] set
+   * to value, for -1 we use the upper bound from
+   * https://arxiv.org/abs/1704.05871
+   * @return value for epsturb
+   */
+  double CalcEpsTurb(double epsturb_in);
+
+  /**
+   * @brief AbsErr absolute error for numerical integration
+   */
+  const double AbsErr = 0;
+
+  /**
+   * @brief RelErr relative error for numerical integration
+   */
+  const double RelErr = 1e-6;
+
+  /**
+   * @brief reduced Hubble constant
+   *
+   */
+  double h = 0.674;
+
+  /**
+   * @brief Calculate peak frequency of GW signal for sound waves
+   */
+  void CalcPeakFrequencySoundWave();
+
+  /**
+   * @brief Calculate peak amplitude of GW signal for sound waves
+   */
+  void CalcPeakAmplitudeSoundWave();
+
+  /**
+   * @brief Calculate peak frequency of GW signal from turbulence
+   */
+  void CalcPeakFrequencyTurbulence();
+
+  /**
+   * @brief Calculate peak amplitude of GW signal from turbulence
+   */
+  void CalcPeakAmplitudeTurbulence();
+
+  /**
+   * @brief Amplitude of GW signal as a function of
+   * @param f frequency
+   * @param swON true = contribution from sound waves switched on,
+   * false = switched off
+   * @param turbON true = contribution from turbulence switched on,
+   * false = switched off
+   * @return h2OmegaGW
+   */
+  double CalcGWAmplitude(double f, bool swON, bool turbON);
+
+  /**
+   * @brief GetSNR
+   * @param fmin minimal frequency
+   * @param fmax maximal frequency
+   * @param T duration of exp. data acquisition, default value: 3 years
+   * @return signal-to-noise (SNR) ratio at LISA
+   */
+  double GetSNR(const double fmin, const double fmax, const double T = 3);
+
+  /**
+   * @brief snr_integrand friend to define inner integrand of SNR integral
+   */
+  friend double snr_integrand(double freq, void *params);
+};
+
+/**
+ * @brief SIfunc
+ * @param f frequency
+ * @return value of SI-function from LISA mission performance requirement for
+ * power spectral density
+ */
+double SIfunc(const double f);
+
+/**
+ * @brief Rfunc
+ * @param f frequency
+ * @return value of R-function from LISA mission performance requirement for
+ * power spectral density
+ */
+double Rfunc(const double f);
+
+/**
+ * @brief powspec_density
+ * @param f frequency
+ * @return value of power spectral density-function from LISA mission
+ * performance requirement
+ */
+double powspec_density(const double f);
+
+double h2OmSens(const double f);
```

Comment:
What does this function do?

Reply:
LISA's nominal sensitivity. Updated the doxygen's description for this function